scripts/stack-trace-parse

```diff
@@ -0,0 +1,195 @@
+        parse_line "${line}"
+    done < "$1"
+    check_state
+}
+
+usage() {
+    echo -e "Usage:"
+    echo -e "\t1. Parse panic stack trace from stored output:"
+    echo -e "\t\t${script_name} [-o <\"opt\">] [-s] -e <executable_path> -f <stack_trace_file>"
+    echo -e "\t2. Parse panic stack trace from output of an enclave over pipe:"
+    echo -e "\t\t<output_stream> | ${script_name} [-o <\"opt\">] [-s] -e <executable_path>"
+    echo -e "\t3. Paste stack trace addresses from an output in raw mode."
+    echo -e "\t\t${script_name} [-o <\"opt\">] [-s] -e <executable_path> -r <enclave_load_address>"
+    echo -e "Note:"
+    echo -e "\t1. -o <\"opt\">:"
+    echo -e "\t\tThis option lets user customize the output format of the stack trace translation."
+    echo -e "\t\tHere, opt indicates options passed to addr2line, default being \"${default_options}\"."
+    echo -e "\t\tPlease refer to manpage of addr2line for further details about options."
+    echo -e "\t\topt must be enclosed in double quotes."
+    echo -e "\t2. -s:"
+    echo -e "\t\tThis option suppresses the output that does not pertain to stack trace of a panic."
+    echo -e "\t\tWithout -s, lines of output before and after the panic stack trace would be echoed on the screen."
+    echo -e "\t3. -r <enclave_load_address>:"
+    echo -e "\t\tThis option lets user run the script in raw mode and parses only stack trace addresses."
+    echo -e "\t\tThis mode expects only back trace as an input."
+    echo -e "\t\tUser could start this script with -r option and copy-paste actual stack trace in stdin, until,"
+    echo -e "\t\tthe script detects EOF (ctrl-d)."
+    echo -e "\t\tAlternatively, in raw mode, a file with a list of only stack addresses (one at the end of each line)"
+    echo -e "\t\tcould be inputted as a file over pipe or with -f option."
+    echo -e "\t\tThe enclave_load_address must be in hexadecimal with a 0x prefix."
+    echo -e "Help:"
+    echo -e "\t${script_name} -h prints help."
+}
+
+detect_term
+while getopts "h?o:e:f:sr:" opt; do
+    case "$opt" in
+    h|\?)
+        usage
+        drain 0
+        ;;
+    e) executable=$OPTARG
+        ;;
+    o) options=$OPTARG
+        ;;
+    f)  trace_file=$OPTARG
+        ;;
+    s) supress=1
+        ;;
+    r)
+        reset_state="printing"
+        start_address=$OPTARG
+        ;; 
+    esac
+done
+
+shift $((OPTIND-1))
+
+if [ ! -z $1 ]; then
+    echo "Extra Arguments detected $@"
+    usage
+    drain -1
+fi
+
+if [ -z ${executable} ]; then
+    echo "No executable file given"
+    usage
+    drain -1
+fi
+
+state=${reset_state}
+
+#Options must begin with - and have f, since we provide the exec path with -f
+if [[ ! $options =~ -.* ]]; then
+    options="-${options}"
+fi
+
+if [ -z ${trace_file} ]; then
+    parse_input
+else
+    if [ "${terminal}" == "1" ]; then
+        parse_file "${trace_file}"
+    else
+        echo "trace file provided and not running on standard input terminal"
```

Comment:
Why is this an error?

Reply:
This happens if the user is doing ./run.sh | enclave-runner/src/loader.rs -e <exe> -f <output_file>
As in there is file provided + output is piped into the script.